apps/carbon/app/hooks/useCustomColumns.tsx

```diff
@@ -0,0 +1,55 @@
+import type { Json } from "@carbon/database";
+import { Enumerable } from "@carbon/react";
+import { useMemo } from "react";
+import type { ColumnDef } from "@tanstack/react-table";
+import { z } from "zod";
+import { path } from "~/utils/path";
+import { useRouteData } from "./useRouteData";
+
+export function useCustomColumns(table: string) {
+  const data = useRouteData<{
+    customFields: { table: string; name: string; fields: Json[] }[];
+  }>(path.to.authenticatedRoot);
+
+  const schema = useMemo<Fields>(() => {
+    let schemaResult: Fields = [];
+    data?.customFields.forEach((field) => {
+      if (field.table === table) {
+        const fields = fieldValidator.safeParse(field.fields);
+        if (fields.success && "table" in field) {
+          schemaResult = fields.data;
+        }
+      }
+    });
+    return schemaResult;
+  }, [data?.customFields, table]);
```

Comment:
I think you can replace all this with:

```suggestion
  const schema = useCustomFieldsSchema(table);
```

Reply:
useCustomFieldsSchema does not take a table arg, should I create another hook that accepts an argument or should I do this instead - 

```
const customFieldsSchema = useCustomFieldsSchema(table);
const tableFields = customFieldsSchema?.[table];
```

Reply:
Sorry --- I think you'd use it like this:

```
const schemas = useCustomFieldsSchema();
const schema = schemas[table];
```